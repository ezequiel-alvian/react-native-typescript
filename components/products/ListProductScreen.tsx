import React, { useEffect } from 'react'
import { View, Text, StyleSheet, TouchableHighlight } from 'react-native'
import { ListProduct, Nav } from '../../types/types'
import { ListItem } from '@rneui/themed'
import TabletPromotion from './TabletPromotion'
import { useNavigation } from '@react-navigation/native'
import { useDispatch } from "react-redux"
import { addTasks } from '../../reducer/productsReducer'

const ListProductScreen = ({item}: { item: ListProduct }) => {
    
    const { navigate } = useNavigation<Nav>()
    const dispatch = useDispatch()

    const onPress = ({item}:{item:ListProduct}) => { 
        navigate('SpecificProduct') 
        dispatch(addTasks(item))
    }
    
    return (
        <ListItem style={styles.cardContainer} onPress={()=> onPress({item})}>
            <ListItem.Content key={item.id}>
                    {item.promotion && <TabletPromotion promotion={item.promotion}/>}
                    <View style={styles.cardDisplay}>
                        <View style={styles.cardImage}></View>
                        <View style={styles.containerText}>
                            <ListItem.Title style={styles.title}>{item.product}</ListItem.Title>
                            <ListItem.Subtitle style={styles.subTitle}>{item.city} - {item.location}</ListItem.Subtitle>
                            <Text style={styles.textColor} >${item.price}</Text>
                            <Text style={styles.text}>{item.amount} kg</Text>
                        </View>
                    </View>
            </ListItem.Content>
        </ListItem>
    )
}


const styles = StyleSheet.create({
    cardDisplay:{
        display: 'flex',
        justifyContent:'flex-start',
        flexDirection:'row',
        flex: 1,
    },
    cardContainer: {
        margin: 10,
        borderColor:'#D3D3D3',
        borderBottomWidth:2,
        flex: 1,
        display:'flex',
        flexDirection:'column'
    },
    cardImage: {
        width: 60,
        height: 60,
        backgroundColor:'#00FF7F',
        marginRight:20,
    },
    containerText:{
        marginBottom:5
    },
    title: {
        fontSize:20,
        fontWeight:'bold',
        fontFamily:'Helvetica'
    },
    subTitle: {
        fontSize:14,
        fontWeight:'400',
        color:'#808080',
        fontFamily:'Helvetica',
        marginBottom:8
    },
    text: {
        fontSize:14,
        fontWeight:'400',
        color:'#808080',
        fontFamily:'Helvetica'
    },
    textColor: {
        fontSize:14,
        fontWeight:'600',
        color:'#000',
        fontFamily:'Helvetica'
    }
})


export default ListProductScreen