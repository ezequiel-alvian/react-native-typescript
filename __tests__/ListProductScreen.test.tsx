import React from 'react'
import { create, act } from 'react-test-renderer'
import ListProductScreen from '../components/products/ListProductScreen'
import data from './data.json' 
import { ListItem } from '@rneui/base'

describe('<ListProductScreen/>', () => {
    const item =data
    const component:any = create(<ListProductScreen item={item}></ListProductScreen>)
    
    test('<ListProductScreen/>', () => {
        expect(component).toMatchSnapshot()
    })

    test('button touch', ()=>{
        const buttonTouch = component.root.findByProps({testID:'myButton'}).props
        console.log(buttonTouch)
        act(()=> buttonTouch.onPress())
    })

})




