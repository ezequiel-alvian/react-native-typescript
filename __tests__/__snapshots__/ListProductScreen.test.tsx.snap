// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListProductScreen/> <ListProductScreen/> 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderBottomWidth": 2,
      "borderColor": "#D3D3D3",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "margin": 10,
    }
  }
  testID="myButton"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
    testID="RNE__LISTITEM__padView"
  >
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      theme={
        {
          "colors": {
            "background": "#ffffff",
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": {
              "android": {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ad1457",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
          "spacing": {
            "lg": 12,
            "md": 8,
            "sm": 4,
            "xl": 24,
            "xs": 2,
          },
        }
      }
    >
      <View
        style={
          {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "width": 710,
          }
        }
      >
        <Text
          style={
            {
              "backgroundColor": "#00FF7F",
              "padding": 5,
            }
          }
        >
          -25%
        </Text>
      </View>
      <View
        style={
          {
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#00FF7F",
              "height": 60,
              "marginRight": 20,
              "width": 60,
            }
          }
        />
        <View
          style={
            {
              "marginBottom": 5,
            }
          }
        >
          <Text
            accessibilityRole="text"
            style={
              {
                "backgroundColor": "transparent",
                "color": "#242424",
                "fontFamily": "Helvetica",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
            testID="listItemTitle"
          >
            Pera
          </Text>
          <Text
            accessibilityRole="text"
            style={
              {
                "backgroundColor": "transparent",
                "color": "#808080",
                "fontFamily": "Helvetica",
                "fontSize": 14,
                "fontWeight": "400",
                "marginBottom": 8,
              }
            }
            testID="listItemTitle"
          >
            Roca
             - 
            Roca
          </Text>
          <Text
            style={
              {
                "color": "#000",
                "fontFamily": "Helvetica",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            $
            600
          </Text>
          <Text
            style={
              {
                "color": "#808080",
                "fontFamily": "Helvetica",
                "fontSize": 14,
                "fontWeight": "400",
              }
            }
          >
            40
             kg
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
